--- unicorn-1.0.1/config.mk.orig	2017-04-20 02:14:24.000000000 -0400
+++ unicorn-1.0.1/config.mk	2017-07-03 10:04:03.852514419 -0400
@@ -8,14 +8,14 @@
 # Compile with debug info when you want to debug code.
 # Change this to 'no' for release edition.
 
-UNICORN_DEBUG ?= yes
+#UNICORN_DEBUG ?= yes
+UNICORN_DEBUG ?= no
 
 ################################################################################
 # Specify which archs you want to compile in. By default, we build all archs.
 
 UNICORN_ARCHS ?= x86 m68k arm aarch64 mips sparc
 
-
 ################################################################################
 # Change 'UNICORN_STATIC = yes' to 'UNICORN_STATIC = no' to avoid building
 # a static library.
